File: arc-layout/src/main/java/com/wkz/arclayout/ClipPathManager.java
package com.wkz.arclayout;

import android.graphics.Paint;
import android.graphics.Path;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public interface ClipPathManager {

    @NonNull
    Path createMask(int width, int height);

    @Nullable
    Path getShadowConvexPath();

    void setupClipLayout(int width, int height);

    Paint getPaint();

    boolean requiresBitmap();
}

File: arc-layout/src/main/java/com/wkz/arclayout/ClipPathManager.kt
package com.wkz.arclayout

import android.graphics.Paint
import android.graphics.Path

interface ClipPathManager {
    fun createMask(width: Int, height: Int): Path
    val shadowConvexPath: Path?
    fun setupClipLayout(width: Int, height: Int)
    val paint: Paint
    fun requiresBitmap(): Boolean
}